modules/nw-sriov-network-attachment.adoc

```diff
@@ -38,45 +54,49 @@ metadata:
   namespace: openshift-sriov-network-operator <2>
 spec:
   networkNamespace: <target_namespace> <3>
-  ipam: <ipam> <4>
+  resourceName: <sriov_resource_name> <4>
   vlan: <vlan> <5>
-  resourceName: <sriov_resource_name> <6>
-  linkState: <link_state> <7>
-  maxTxRate: <max_tx_rate> <8>
-  minTxRate: <min_rx_rate> <9>
-  vlanQoS: <vlan_qos> <10>
-  spoofChk: "<spoof_check>" <11>
+  spoofChk: "<spoof_check>" <6>
+ifdef::ocp-sriov-net[]
+  ipam: <ipam> <7>
+  linkState: <link_state> <8>
+  maxTxRate: <max_tx_rate> <9>
+  minTxRate: <min_rx_rate> <10>
+  vlanQoS: <vlan_qos> <11>
   trust: "<trust_vf>" <12>
   capabilities: <capabilities> <13>
+endif::ocp-sriov-net[]
 ----
 <1> Replace `<name>` with a name for the CR. The SR-IOV Network Operator creates a NetworkAttachmentDefinition CR with same name.
 <2> Specify the namespace where the SR-IOV Operator is installed.
```

Comment:
Please add if it is optional, mandatory and also default values if not configured.
It looks to be mandatory.
if we don't specify it will create sriovnetwork under "default" namespace.

example :

# cat snet.yaml
---
apiVersion: sriovnetwork.openshift.io/v1
kind: SriovNetwork
metadata:
  name: sriov-gkapoor-101
spec:
  ipam: |
    {}
  vlan: 0
  spoofChk: "on"
  resourceName: test_gkapoor11
  networkNamespace: testing1234
=========================================
# oc get sriovnetworks -A
NAMESPACE                          NAME                AGE
default                            sriov-gkapoor-101   154m

Reply:
@geetikakay - is is possible that the network was created in "default" namespace because the 'networkNamespace' that was provided (testing1234) does not exist (as per your other comment)? Since that should provide the target namespace for the network.
Seems odd that not specifying the namespace for the SRIOV operator would result in this being created in the "default" namespace.

Reply:
@aburdenthehand  right. I am not sure either about this behaviour. Suggestions whom should we reach?

Reply:
Good question. Perhaps @phoracek and @zshi-redhat ?

Reply:
You can create the SriovNetwork in any namespace. However, it won't be acted upon (it won't result in NetworkAttachmentDefinition created) unless it is in the namespace of the SR-IOV operator AFAIK. So it is mandatory to set the namespace to operator's namespace.

Reply:
If spec.NetworkNamespace is not provided in sriovnetwork CR, the default behavior in SR-IOV Operator is to use the namespace of sriovnetwork CR, which in above case (I assume) is `default`.

If you configure sriovnetwork CR namespace to another one (e.g. openshift-sriov-network-operator), then net-attach-def will be created in `openshift-sriov-network-operator` namespace.

Reply:
Thanks folks. I've updated this module quite a bit to include the info you've provided/clarified here, and with the ipam bug. It's a bit messy with conditionals so it might be easier to look at the preview builds for CNV and OCP for a final review:
CNV: https://cnv-2160-sriov--ocpdocs.netlify.app/openshift-enterprise/latest/virt/virtual_machines/vm_networking/virt-defining-an-sriov-network.html
OCP: https://cnv-2160-sriov--ocpdocs.netlify.app/openshift-enterprise/latest/networking/hardware_networks/configuring-sriov-net-attach.html